src/Stratis.Bitcoin/P2P/SelfEndpointTracker.cs

```diff
@@ -11,6 +19,15 @@ public class SelfEndpointTracker : ISelfEndpointTracker
         /// <summary>Hashset to hold the endpoints currently known to be itself.</summary>
         private readonly ConcurrentHashSet<IPEndPoint> knownSelfEndpoints = new ConcurrentHashSet<IPEndPoint>();
 
+        /// <summary>External IP address of the node.</summary>
+        public IPEndPoint MyExternalAddress { get; set; }
+
+        /// <summary>Whether IP address of the node is final or can be updated.</summary>
+        public bool IsMyExternalAddressFinal { get; set; } = false;
+
+        /// <summary>Peer score of external IP address of the node.</summary>
+        public int MyExternalAddressPeerScore { get; set; }
```

Comment:
Cool you fixed most of the issues I was going to comment on. I like this much better than the KeyValuePair.  Only small thing I noticed is I think ```MyExternalAddressPeerScore``` can be made private as I don't think it's referenced outside of this class.

Reply:
This should also be private field